staging/src/k8s.io/apimachinery/pkg/util/net/interface_windows.go

```diff
@@ -0,0 +1,86 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package net
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+func chooseHostInterface(addressFamilies AddressFamilyPreference) (net.IP, error) {
+	var nw networkInterfacer = networkInterface{}
+	interfaces, err := getDefaultRouteInterfaces()
+	if err != nil {
+		return chooseIPFromHostInterfaces(nw, addressFamilies)
+	}
+	for _, family := range addressFamilies {
+		for _, intf := range interfaces {
+			ip, err := getIPFromInterface(intf, family, nw)
+			if err != nil {
+				return nil, err
+			}
+			if ip != nil {
+				klog.V(4).Infof("Found active IP %v ", ip)
+				return ip, nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("unable to find suitable IP")
+}
+
+// getDefaultRouteInterfaces returns the interfaces for the IPv4 and IPv6 default routes.
+func getDefaultRouteInterfaces() ([]*net.Interface, error) {
+	// https://docs.microsoft.com/en-us/powershell/module/nettcpip/get-netroute?view=windowsserver2019-ps#example-4--get-the-next-hop-for-the-default-route
+	// -DestinationPrefix values of "0.0.0.0/0" and "::0/0", return entries with the NextHop value being the default
+	// gateway for IPv4, and IPv6, respectively.
+	cmd := exec.Command("powershell", "/c", `Get-NetRoute -ErrorAction Ignore -DestinationPrefix ("0.0.0.0/0","::0/0") | Format-Table -Property "ifIndex","NextHop" -HideTableHeaders`)
```

Comment:
it would be preferable to use golang to avoid creating a process by calling out to powershell if possible

Reply:
Agreed, changed to using the [getipforwardentry2](https://docs.microsoft.com/en-us/windows/win32/api/netioapi/nf-netioapi-getipforwardentry2) function.